docs/en/week11/11-2.md

```diff
@@ -0,0 +1,382 @@
+
+We use the additional term to ensure the transcription is consistent with the rules of the language. Else we may get grammatically wrong transcription.
+
+
+## Beam Search
+
+While returning an output sequence, we can follow the greedy approach, where we take the maximum value of $$P(\vect{y_{t}} \mid \vect{y_{t-1}}...\vect{y_{1}})$$
+
+However, we can end up missing out a good sequence, which may not have the maximum value of $P(\vect{y_{t}} \mid ...)$, as illustrated by the example below.
+
+<!--![Shortest Path failing in the Greedy Approach](https://i.imgur.com/DX8j0Sv.png) -->
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/greedy.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 1:</b> Greedy Approach : Less Optimal Solution<br>
+<br>
+</center>
+
+To remedy this, we employ beam search. Essentially, we consider maximum $k$ tokens in terms of probability at each step $t$ of the sequence. For each of these n-grams, we proceed further and find out the maximum. 
+
+The illustration below shows how Beam Search can lead to a better sequence.
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/bs3.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 2:</b> Beam Search : Stage 1 <br>
+<br>
+</center>
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/bs1.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 3:</b> Beam Search : Stage 2<br>
+<br>
+</center>
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/bs2.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 4:</b> Beam Search : Stage 3<br>
+<br>
+</center>
+
+
+## Graph Transformer Networks
+
+We have previously seen Weighted Finite State Automata (WFSA) being used to represent the alignment graphs, as shown before.
+
+
+Graph Transformer Networks (GTNs) are basically WSFA with automatic differentiation.
+
+Lets look at key differences between Neural Networks (NNs) and GTNs
+
+
+
+|  | NN | GTN |
+| -------- | -------- | -------- |
+| Core Data Structure     | Tensor     | Graph/WFSA     |
+|     | Matrix Multiplcation     | Compose     |
+| Core Operations  | Reduction operations (Sum, Product)| Shortest Distance (Viterbi,Forward) |
+| | Negate, Add, Subtract, ...     | Closure, Union, Concatenate, ...    |
+
+
+## Why Differentiable WSFAs?
+
+* Encode Priors: easy to encode priors into a graph
+* End-to-end: Bridge the training and test process 
+* Facilitate research: When we separate the data from the code it makes it easier to explore different ideas. In our case graph is the data and the code is operations allows us to explore different ideas by changing the graph.
+
+
+## Sequence Criteria with WFSAs
+
+It turns out that lots of the sequence criteria like connection temporal classification (CTC) can be specified as the difference between two forward scores.
+
+* Graphs A: function of input X (speech) and target "output" Y (transription).
```

Comment:
Use `$` for math, here and everywhere else.

Reply:
Just to confirm the whole document 11-2.md should have $ instead of $$ correct? Because in the contribution guide it says that we should use $$.